### PR TITLE
wl: Remove wl_surface_frame as part of wpe_view_data

### PR DESCRIPTION
**Pull request by psaavedra**
The wl_surface_frame callback has all that it needs from the context. It also doesn't requires a reference in wpe_view_data since it is auto destroyed during the callback action.

Also formatting fixes.

### CODE REVIEW

File: platform/wayland/cog-platform-wl.c [hunk 1]
```diff
@@ -1460,10 +1455,8 @@ static const struct wp_presentation_feedback_listener presentation_feedback_list
 static void
 request_frame(void)
 {
-    if (wpe_view_data.frame_callback == NULL) {
-        wpe_view_data.frame_callback = wl_surface_frame(win_data.wl_surface);
-        wl_callback_add_listener(wpe_view_data.frame_callback, &frame_listener, NULL);
-    }
+    struct wl_callback *frame_callback = wl_surface_frame(win_data.wl_surface);
+    wl_callback_add_listener(frame_callback, &frame_listener, NULL);
```
Comment (aperezdc) [inline platform/wayland/cog-platform-wl.c]: Removing the variable and the null-check can result in more than one frame callback being registered for the same frame. The check was there to allow calling `request_frame()` from different places but still having only one callback registered for each frame.

Are you sure that without the check we never schedule more than one callback per frame?

**Comment by aperezdc** [comment]
I would remove the commit with all the formatting fixes, and only do the minimum amount of formatting required by the changes in the other patch. The only formatting change needed to keep the style checker happy seems to be this one:

```diff
--- platform/wayland/cog-platform-wl.c	(before formatting)
+++ platform/wayland/cog-platform-wl.c	(after formatting)
@@ -305,7 +305,7 @@
 static struct {
     struct wpe_view_backend *backend;
     struct wpe_fdo_egl_exported_image *image;
-    struct wl_buffer *buffer;
+    struct wl_buffer                  *buffer;
     bool should_update_opaque_region;
 } wpe_view_data = {
     .should_update_opaque_region = true, /* Force initial update. */
```

**Comment by aperezdc** [comment]
@psaavedra Ping 🏓, please let me know about the question above when you have time. Thanks!

**Comment by psaavedra** [comment]
This PR is obsolete. Closing it.

**Pull request closed without merging**